src/store/modules/nft.js

```diff
@@ -353,29 +362,72 @@ const actions = {
     }
     return owners;
   },
-  async fetchNFTListByAddress({ commit }, address) {
+  async fetchNFTListByAddress({ commit, dispatch }, address) {
     const [collectedNFTs, createdNFTClasses] = await Promise.all([
       getNFTsRespectDualPrefix(this.$api, address),
       getNFTClassesRespectDualPrefix(this.$api, address),
     ]);
-    const timestampMap = {};
-    collectedNFTs.forEach(nft => {
-      const { classId, timestamp } = nft;
-      if (!timestampMap[classId] || timestampMap[classId] < timestamp) {
-        timestampMap[classId] = timestamp;
-      }
+
+    const nftClassIdDataMap = new Map();
+    collectedNFTs.forEach(n => nftClassIdDataMap.set(n.class_id, n.class_data));
+    createdNFTClasses.forEach(c => nftClassIdDataMap.set(c.id, c));
+    nftClassIdDataMap.forEach((classData, classId) => {
+      dispatch('formatAndSetNFTClassMetadata', { classId, classData });
     });
+
+    const formatedCreatedNFTClasses = createdNFTClasses.map(formatNFTClassInfo);
+    const formatedCollectedNFTs = collectedNFTs.map(formatNFTInfo);
     commit(TYPES.NFT_SET_USER_CLASSID_LIST_MAP, {
       address,
       nfts: {
-        created: normalizeNFTList(createdNFTClasses),
-        collected: normalizeNFTList(collectedNFTs),
+        created: normalizeNFTList(formatedCreatedNFTClasses, getters),
+        collected: normalizeNFTList(formatedCollectedNFTs, getters),
```

Comment:
I think normalizeNFTList can be part of the getters, so you dont have to include getters as params

Reply:
If `getNormalizeNFTList()` is a function from getters, can we use it before committing data into the store?

Reply:
I think it is no problem at all, and I think can keep `normalizeNFTList` for the name of the getter